openedx/core/djangolib/tests/test_blockstore_cache.py

```diff
@@ -23,9 +26,8 @@ def setUpClass(cls):
         cls.draft = api.get_or_create_bundle_draft(cls.bundle.uuid, draft_name="test-draft")
 
 
-@unittest.skipUnless(settings.RUN_BLOCKSTORE_TESTS, "Requires a running Blockstore server")
 @patch('openedx.core.djangolib.blockstore_cache.MAX_BLOCKSTORE_CACHE_DELAY', 0)
-class BundleCacheTest(TestWithBundleMixin, unittest.TestCase):
```

Comment:
> Switching between using the blockstore standalone service and the blockstore installed app causes an issue with the [blockstore cache for the bundle file URLs](https://github.com/openedx/edx-platform/blob/db32ff2cdf678fa8edd12c9da76a76eef0478614/openedx/core/djangolib/blockstore_cache.py#L160-L235).
In the testing instructions, I've opted to simply clear the caches, but another approach is to use the waffle switch: blockstore.use_blockstore_app_api setting as part of the cache key. I'd welcome input on this decision.

This is only needed during devstack because the storage urls change, right? Or is there something else that we need to clear the cache for?

Also for some reason restarting the lms or studio isn't clearing the caches for me. I have to use the `blockstore_cache.cache.clear()` for it to work.

Reply:
@symbolist Yes, this is only because the storage URLs change -- everything else in the cache is fine.
I had an idea that maybe I could use the `blockstore.use_blockstore_app_api` waffle switch setting as part of the cache key for these files, but I worried this might slow down the cache access.

> Also for some reason restarting the lms or studio isn't clearing the caches for me. I have to use the `blockstore_cache.cache.clear()` for it to work.

Hmm.. I started off by doing that, but didn't get results until I restarted LMS/Studio. I'll add that to the PR description just in case.

Reply:
Ah, good point! I'll update the PR description to reflect this.